### ISSUE TITLE
Incorrect typedef ["call-signature"] for constructors

### ISSUE THREAD

**Issue by abeggchr** [palantir/tslint#358]
```typescript
class Greeter {
    greeting: string;
    constructor(message: string) {
        this.greeting = message;
    }
}
```

triggers the `typedef ["call-signature"] ` rule on the constructor declaration, although type annotations cannot appear on a constructor declaration.

I suspect a versioning problem. I am using TypeScript 1.4 and TSLint shipped with WebEssentials 2.5 (must be the latest TSLint version from around 17-NOV-14).

**Comment by adidahiya** [comment]
Yes, that sounds like a versioning problem. Please try upgrading to the latest tslint v2.1.1. Anything pre-2.0 is not ready to work with TS 1.4.

**Comment by abeggchr** [comment]
Thanks. WebEssentials 2.5.4 actually ships with TSLint 1.0.1.
